include/deal.II/lac/petsc_block_vector.h

```diff
@@ -85,13 +87,13 @@ namespace PETScWrappers
      * of different sizes.
      */
     explicit BlockVector (const unsigned int num_blocks = 0,
-                          const size_type    block_size = 0);
+                          const size_type    block_size = 0) DEAL_II_DEPRECATED;
```

Comment:
can't you just deprecate the whole class? That might be simpler than deprecating the constructors. Same with the other classes.

I think the correct syntax would be
```
  class BlockVector {
     ...
  } DEAL_II_DEPRECATED;
```